### PR TITLE
Merge VectorHardSigmoid for x86 and ARM

### PR DESCRIPTION
**Pull request by FedyuninV**

### CODE REVIEW

File: NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h [hunk 1]
```diff
@@ -910,6 +910,40 @@ inline void vectorHSwish( const float* first, float* result, int vectorSize )
 	}
 }
 
+//------------------------------------------------------------------------------------------------------------
+
+inline float32x4_t vectorHardSigmoidWorker( const float32x4_t& val,
+	const float32x4_t& zero, const float32x4_t& one, const float32x4_t& slope, const float32x4_t& bias )
+{
+	float32x4_t mainVal = vaddq_f32( vmulq_f32( val, slope ), bias );
```
Comment (lordnn) [inline NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h:918]: It's enough scalar values for `slope` and `bias` (use vmulq_n_f32 and vaddq_n_f32 intrinsics)

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h [hunk 2]
```diff
@@ -1174,6 +1174,45 @@ inline void vectorHSwish( const float* first, float* result, int vectorSize )
...
+			value = _mm_add_ps( value, biasSse );
+			_mm_storeu_ps(result, _mm_min_ps( _mm_max_ps( value, zeroSse ), oneSse ) );
+
+			first += 4;
+			result += 4;
+		}
+	}
+
+	const float maxXValue = ( 1 - bias ) / slope;
+	const float minXValue = -bias / slope;
```
Comment (lordnn) [inline NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h]: I would have expect different results between vectorized and scalar versions...

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h [hunk 3]
```diff
@@ -1174,6 +1174,45 @@ inline void vectorHSwish( const float* first, float* result, int vectorSize )
 	}
 }
 
+//------------------------------------------------------------------------------------------------------------
+
+inline void vectorHardSigmoid( const float* first, float* result, float slope, float bias, int vectorSize )
+{
+	int sseSize;
+	int nonSseSize;
+	checkSse( vectorSize, sseSize, nonSseSize );
```
Comment (lordnn) [inline NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h]: I don't understand your passion about `checkSse()` and `GetCount4()`. Look at #858

**Comment by lordnn** [comment]
Some copyrights was outdated.

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h (hunk 3 above)
Comment (FedyuninV) [reply NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h]: It's not a passion, it's laziness)
This code was moved from the other file as-is.

File: NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h (hunk 1 above)
Comment (FedyuninV) [reply NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h:918]: Right now we don't use MVE/Helium.

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h (hunk 2 above)
Comment (FedyuninV) [reply NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h]: Agree. Fixed.

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h (hunk 3 above)
Comment (FedyuninV) [reply NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h]: Btw, fixed

**Comment by FedyuninV** [comment]
> Some copyrights was outdated.

Fixed

File: NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h (hunk 1 above)
Comment (lordnn) [reply NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h:918]: It's not MVE-specific, IMHO.

[vmulq_n_f32](https://developer.arm.com/architectures/instruction-sets/intrinsics/vmulq_n_f32)
Hmm, but there is no vaddq_n_f32. :(

File: NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h (hunk 3 above)
Comment (lordnn) [reply NeoMathEngine/src/CPU/x86/CpuX86MathEngineVectorMathPrivate.h]: What about `GetCount4()`?

File: NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h (hunk 1 above)
Comment (FedyuninV) [reply NeoMathEngine/src/CPU/arm/CpuArmMathEngineVectorMathPrivate.h:918]: It's unexpected that `vmulq_n_f32` is NEON but `vaddq_n_f32` is VME. I've checked only `vaddq_n_f32` and concluded that `mul` is in the same set.

**Pull request merged**